web-common/src/main/java/edu/unc/lib/boxc/web/common/services/AccessCopiesService.java

```diff
@@ -97,6 +101,49 @@ public boolean hasViewableFiles(ContentObjectRecord briefObj, AccessGroupSet pri
         return resp.getResults().getNumFound() > 0;
     }
 
+    /**
+     * Returns true if the file is a pdf and is the primary object or the object has no primary object
+     * @param workObjs
+     * @return
+     */
+    public boolean pdfViewerNeeded(List<ContentObjectRecord> workObjs) {
```

Comment:
I think this implementation might run into issues if the work has more than one page worth of files and the primary object isn't on the first page. But I think there are a few performance improvements we can make at the same time.

You should only need to pass the workObject itself into this method, rather than querying for all its children beforehand. We don't need to query for children if the work uses the UV (this method doesn't need to be called in that case), and we don't need to query for them if the Work itself passes `hasPdf` (which implies that the primaryObject is a pdf). 

So that leaves us with works without primary objects. For that scenario, I believe you can query for the first child (so rows=1). Then you can check `searchResult.getResultCount()` to get the total number of results for the query, and if it is > 1 then its a multi-file work and you would want to return false. If its == 1, then you would return true if hasPdf passes for that child.

I really wish this class had a unit test... It might be worth while to add one to be honest. Let me know if I can help with doing so, i just suspect it will make writing this functionality easier